depends: Override default build type for `libevent`

hebasto opened the pull request:
This PR fixes a regression for the `libevent` package introduced in https://github.com/bitcoin/bitcoin/pull/29835.

The `libevent` package defaults to the "Release" build type, which overrides our per-build-type optimization flags with `-O3`.

To prevent this behavior, set `CMAKE_BUILD_TYPE` to "None", consistent with how other packages are handled.

hebasto commented:
My Guix build:
```
aarch64
86fcefb01cf8fb3d3bfc0639b270f4df96567066620cdaf9d1a3fc70664c1755  guix-build-dd7868a173fb/output/aarch64-linux-gnu/SHA256SUMS.part
ebca518cd64e7f45d96a758ed8d06c3a2689aa1783f75366e3a5f4a32fd79bca  guix-build-dd7868a173fb/output/aarch64-linux-gnu/bitcoin-dd7868a173fb-aarch64-linux-gnu-debug.tar.gz
b93ef69b68418482fab8977b5a590fbacd4aa37ae5d8f1da1cc5b071df9b6c17  guix-build-dd7868a173fb/output/aarch64-linux-gnu/bitcoin-dd7868a173fb-aarch64-linux-gnu.tar.gz
62f77252880738db554f7977e4f444cf28f9a71d2a3e0055e1efebc5fca65816  guix-build-dd7868a173fb/output/arm-linux-gnueabihf/SHA256SUMS.part
07a7bc7813d9fb0f3cf881081059ca02f744e172c6c9c9f5bdd5254a610e70b5  guix-build-dd7868a173fb/output/arm-linux-gnueabihf/bitcoin-dd7868a173fb-arm-linux-gnueabihf-debug.tar.gz
4e5f7b781379e5decbb61192b7862eb94736b0af02fcd4ce00b2c4709a74ddc5  guix-build-dd7868a173fb/output/arm-linux-gnueabihf/bitcoin-dd7868a173fb-arm-linux-gnueabihf.tar.gz
5b61db8f0ba10d57bf1f2474d5b5a36f5b3fe95c74d57a3912deb2603361a728  guix-build-dd7868a173fb/output/arm64-apple-darwin/SHA256SUMS.part
18a51ddf6355c85cb90e9acafd226317c72e7b2fe6192c81661b74f5408dde2a  guix-build-dd7868a173fb/output/arm64-apple-darwin/bitcoin-dd7868a173fb-arm64-apple-darwin-unsigned.tar.gz
cf0758ab71fe90d7ba71f48f2513339cf7564cd5739f37eebf9b4f2af90d2063  guix-build-dd7868a173fb/output/arm64-apple-darwin/bitcoin-dd7868a173fb-arm64-apple-darwin-unsigned.zip
e37bb992236de3b84eaad91c6ddda6aa6dc07639880615074cd22124534a8da6  guix-build-dd7868a173fb/output/arm64-apple-darwin/bitcoin-dd7868a173fb-arm64-apple-darwin.tar.gz
7aa97c05988b8ef8d52d2564c2a0a44e612f60aeb2112743dee27bc0dddb3506  guix-build-dd7868a173fb/output/dist-archive/bitcoin-dd7868a173fb.tar.gz
cdbc38e6fe5a92f25085ab8102e4e89ec5b176e0022685ee5f68f386de231adb  guix-build-dd7868a173fb/output/powerpc64-linux-gnu/SHA256SUMS.part
b524a5d6fd5a55c036e70b7d6c61c6d880cfe6aab5ed41e14064a57ce9a1b4de  guix-build-dd7868a173fb/output/powerpc64-linux-gnu/bitcoin-dd7868a173fb-powerpc64-linux-gnu-debug.tar.gz
16e96e5e3b314fb63638db2ade0039c73df53cf74dc09918434cc6d96dc38334  guix-build-dd7868a173fb/output/powerpc64-linux-gnu/bitcoin-dd7868a173fb-powerpc64-linux-gnu.tar.gz
f10b60add22f1680c52de6a920486c7f742010a66fb224586ce4fe6550064fb2  guix-build-dd7868a173fb/output/riscv64-linux-gnu/SHA256SUMS.part
c6cc34f9b81e134e5b4707d89983238ce25c88248e890193c97899a95c741448  guix-build-dd7868a173fb/output/riscv64-linux-gnu/bitcoin-dd7868a173fb-riscv64-linux-gnu-debug.tar.gz
f1d5d2fad90a3e6eeb47fd094671c59be290ab414ab79623f456596724d1e949  guix-build-dd7868a173fb/output/riscv64-linux-gnu/bitcoin-dd7868a173fb-riscv64-linux-gnu.tar.gz
7077ed544d94572889afde0bc29bb035010b1a8a5eb3c4c09c9cf8225227610b  guix-build-dd7868a173fb/output/x86_64-apple-darwin/SHA256SUMS.part
0c2d359c91089927ae758e00b1b9ddc59ef897e2ff86c3dbb7ede8ec5379805b  guix-build-dd7868a173fb/output/x86_64-apple-darwin/bitcoin-dd7868a173fb-x86_64-apple-darwin-unsigned.tar.gz
dc57e5974dced23a30ade9a4f98fe33d1d31120f314fff691ee2676473d5cde8  guix-build-dd7868a173fb/output/x86_64-apple-darwin/bitcoin-dd7868a173fb-x86_64-apple-darwin-unsigned.zip
bd625e3808707f92f8e6977fefd263b289cfe72b9bd973e8b08eae9d64ec7774  guix-build-dd7868a173fb/output/x86_64-apple-darwin/bitcoin-dd7868a173fb-x86_64-apple-darwin.tar.gz
2080821630d89152a1a583a4f87c6d605ebda367eab7691963dfd9af2c67b372  guix-build-dd7868a173fb/output/x86_64-linux-gnu/SHA256SUMS.part
a4f783b61322c260dc2770d4dbc1033f622f5ca8ffd21c7ca82fa5ca359b713d  guix-build-dd7868a173fb/output/x86_64-linux-gnu/bitcoin-dd7868a173fb-x86_64-linux-gnu-debug.tar.gz
68f1f9f17e14ea3e3486a13e6282d12e399ee58d4556e3d9e512863d3f777bce  guix-build-dd7868a173fb/output/x86_64-linux-gnu/bitcoin-dd7868a173fb-x86_64-linux-gnu.tar.gz
e11b5fc3884d7552b9b54fade3b5e17926b47e10818012adc1de420585fbcb29  guix-build-dd7868a173fb/output/x86_64-w64-mingw32/SHA256SUMS.part
c8cd6e79b311eb71b4ef334c3a0c9fb3823bded3047ee53b21177babc9eb36aa  guix-build-dd7868a173fb/output/x86_64-w64-mingw32/bitcoin-dd7868a173fb-win64-debug.zip
174a3d0863e32ec1f5bfb40d63c2ed29e49d1ee4d6714dda90f33badcbd67951  guix-build-dd7868a173fb/output/x86_64-w64-mingw32/bitcoin-dd7868a173fb-win64-setup-unsigned.exe
4441e572ede42a909338f69669f9d8c4adb16a59d1c05e14ed60f4b0d33f6bc4  guix-build-dd7868a173fb/output/x86_64-w64-mingw32/bitcoin-dd7868a173fb-win64-unsigned.tar.gz
d256b7e9d625664d3b274b74a02eff10a084477c246a52f19a42f24590b936d5  guix-build-dd7868a173fb/output/x86_64-w64-mingw32/bitcoin-dd7868a173fb-win64.zip
```

theuni commented:
> For packages that do not set a default build type, this change does not affect their behaviour.

Are you sure this is true? It definitely _could_ affect behavior. I'm wondering if we should do it per-package, even if it's every package, as a sign-off that we've actually checked that it does the right thing.

hebasto commented:
> > For packages that do not set a default build type, this change does not affect their behaviour.
> 
> Are you sure this is true? It definitely _could_ affect behavior. I'm wondering if we should do it per-package, even if it's every package, as a sign-off that we've actually checked that it does the right thing.

Reworked.

hebasto commented:
My Guix build:
```
aarch64
9642c7c336c6dda11fd7f0e085910ab9f4edf73d41401a6df333f7ac979fd845  guix-build-d7d3bb1bbcc0/output/aarch64-linux-gnu/SHA256SUMS.part
f5f224be7a011f1465f995bb21a9f3af18567911272d795a937913370012d83f  guix-build-d7d3bb1bbcc0/output/aarch64-linux-gnu/bitcoin-d7d3bb1bbcc0-aarch64-linux-gnu-debug.tar.gz
7cd81a1879b84a00a9a623f056f8f06127377cedd87220d577a32a81beda79de  guix-build-d7d3bb1bbcc0/output/aarch64-linux-gnu/bitcoin-d7d3bb1bbcc0-aarch64-linux-gnu.tar.gz
1d75c0d8ea2b82580a82b03879de60adcdfdea796b97fa5dbd16ddfcbf6b5b30  guix-build-d7d3bb1bbcc0/output/arm-linux-gnueabihf/SHA256SUMS.part
5596badf7fc9fd750ad1bd136e0c5d64e008fcc11b57c42f1644102e9b8e85e7  guix-build-d7d3bb1bbcc0/output/arm-linux-gnueabihf/bitcoin-d7d3bb1bbcc0-arm-linux-gnueabihf-debug.tar.gz
b446426ac00ceb737ab62f5cd807f43f0e15efeb3f0ac36faede49d3f6870890  guix-build-d7d3bb1bbcc0/output/arm-linux-gnueabihf/bitcoin-d7d3bb1bbcc0-arm-linux-gnueabihf.tar.gz
1ec9474c4d4ba7b8e09a6ee61b9cb11b76c12cee2d164dbaf6586563212ec7ac  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/SHA256SUMS.part
bab59243ae46d748397f9aadea84a2673381527d4f7d57b37de99467f7d93378  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/bitcoin-d7d3bb1bbcc0-arm64-apple-darwin-unsigned.tar.gz
9b926a7e3bcd43c363d42489a0dba92e1b54e7fe015c312de2308a4236e4dd0c  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/bitcoin-d7d3bb1bbcc0-arm64-apple-darwin-unsigned.zip
224ba4173858a7140fa8b96ab4904f8f746091d8cdef40c1d68ff37fc6abb2fb  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/bitcoin-d7d3bb1bbcc0-arm64-apple-darwin.tar.gz
500751e3817551bbdb1377b7947bb666a34d018a018f01b5e4b035e221950f7c  guix-build-d7d3bb1bbcc0/output/dist-archive/bitcoin-d7d3bb1bbcc0.tar.gz
63e08c1f38f893b7b928ee707f8206782c800cc25cf84146f42c9f92c39f43bb  guix-build-d7d3bb1bbcc0/output/powerpc64-linux-gnu/SHA256SUMS.part
8d6efeb86bad556ee510ddf3c0fac0c15cb257e884a057dd5f51c92dd4689653  guix-build-d7d3bb1bbcc0/output/powerpc64-linux-gnu/bitcoin-d7d3bb1bbcc0-powerpc64-linux-gnu-debug.tar.gz
b9c1e9816e0399b5960b0a7f1ed547ae82c839c6d9dc4ec78ffc661e5f4f3db5  guix-build-d7d3bb1bbcc0/output/powerpc64-linux-gnu/bitcoin-d7d3bb1bbcc0-powerpc64-linux-gnu.tar.gz
d09c2c663f856cabbad0259fe200cd6091cc3546647ec77f5517b34bcb9d4429  guix-build-d7d3bb1bbcc0/output/riscv64-linux-gnu/SHA256SUMS.part
0de69c2c507243f2e0da67092e192edf6bb793aa51cbe9567e2e8271d6b8c249  guix-build-d7d3bb1bbcc0/output/riscv64-linux-gnu/bitcoin-d7d3bb1bbcc0-riscv64-linux-gnu-debug.tar.gz
589b6f2646a70cf13443b9011cc5c9d88f4eed78460adfe4f430a8e38ae24df2  guix-build-d7d3bb1bbcc0/output/riscv64-linux-gnu/bitcoin-d7d3bb1bbcc0-riscv64-linux-gnu.tar.gz
f122bd36048306e0b1dad07b0c482582309dad86c01c4c8c02496b90c16bb3ae  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/SHA256SUMS.part
1fc0e8d9ce6e9f6bcc20225a4f49cc494b02178244e18c8176aa3efb0984b305  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/bitcoin-d7d3bb1bbcc0-x86_64-apple-darwin-unsigned.tar.gz
d90db4f98e11304af5340480eeb2d926bf788636d9df6d4f47dafb38842af78f  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/bitcoin-d7d3bb1bbcc0-x86_64-apple-darwin-unsigned.zip
8084a814ec7409a0e209f9049207e304215b5366e67661068398514a93c3bc13  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/bitcoin-d7d3bb1bbcc0-x86_64-apple-darwin.tar.gz
ccf14469e17c447a40ea928944ca0f700d705ab87eb7dea42da5b78a645aeef7  guix-build-d7d3bb1bbcc0/output/x86_64-linux-gnu/SHA256SUMS.part
0e90253dbda704d0877515cfd95b33a0bf289fe6465002bf975b7f7c34a6779a  guix-build-d7d3bb1bbcc0/output/x86_64-linux-gnu/bitcoin-d7d3bb1bbcc0-x86_64-linux-gnu-debug.tar.gz
a3f6985a8483b3b10a4382f185f34bd7c896a174cd2b980dfdf17627c8666135  guix-build-d7d3bb1bbcc0/output/x86_64-linux-gnu/bitcoin-d7d3bb1bbcc0-x86_64-linux-gnu.tar.gz
f8bcaaff159238a6a06e9fe57d03984d4b1d97c0a23818809ef248e2290dc49a  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/SHA256SUMS.part
057d47f78dbdf1efe0809e1172a3cc86797d925bcd2637e264b6d0b4d3777592  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64-debug.zip
5579557e8ddcf4f6ab3b544c8a7905520abe1ab7a626f4aadaebf05098b76f9c  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64-setup-unsigned.exe
13d3e6570cd4da105f1c13bd11ac8596d773b95d2f3cbfb23fd8e42c0a771bcc  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64-unsigned.tar.gz
a6ff418aec9c1d50c758152394e4807f6d989fd94891f6e5f0c29a882dcbf542  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64.zip
```

fanquake reviewed (approved):
~~ACK d7d3bb1bbcc0d9b03ac1d3b4e3a6a537fcff077d~~

fanquake commented:
Spoke too soon. This change is not deterministic.

Guix Build (aarch64):
```bash
d5d3cf8a8cec671bccc5d1dee03f05f67dee52dfd543295d93e71b1338ae6abd  guix-build-d7d3bb1bbcc0/output/aarch64-linux-gnu/SHA256SUMS.part
f5f224be7a011f1465f995bb21a9f3af18567911272d795a937913370012d83f  guix-build-d7d3bb1bbcc0/output/aarch64-linux-gnu/bitcoin-d7d3bb1bbcc0-aarch64-linux-gnu-debug.tar.gz
9e2e815f19be9f7e8480ce5c5bd2443b703e5134667ed3feb57fd2cd75ad9dae  guix-build-d7d3bb1bbcc0/output/aarch64-linux-gnu/bitcoin-d7d3bb1bbcc0-aarch64-linux-gnu.tar.gz
5d0abf903220760a6652409c9be75b4a0adcd21055778958b0ca6e254cb0d3b6  guix-build-d7d3bb1bbcc0/output/arm-linux-gnueabihf/SHA256SUMS.part
5596badf7fc9fd750ad1bd136e0c5d64e008fcc11b57c42f1644102e9b8e85e7  guix-build-d7d3bb1bbcc0/output/arm-linux-gnueabihf/bitcoin-d7d3bb1bbcc0-arm-linux-gnueabihf-debug.tar.gz
6cc5d27e5472d20e9c367506c13b4e41be2252ac3de9643da8179ba2567cd1c3  guix-build-d7d3bb1bbcc0/output/arm-linux-gnueabihf/bitcoin-d7d3bb1bbcc0-arm-linux-gnueabihf.tar.gz
5e5c98ef215004ddc0b5415f56627138e94e53d958a3c00bebc76f38f377c3e5  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/SHA256SUMS.part
8cc86debaee483678477092b40693c6f27acd2c58e4df2654327ab0b9583bbae  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/bitcoin-d7d3bb1bbcc0-arm64-apple-darwin-unsigned.tar.gz
241be8a303e9111aee4cf0617a919ff476fb83c82a3345146c77a397069b94bd  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/bitcoin-d7d3bb1bbcc0-arm64-apple-darwin-unsigned.zip
0d96aee779b9062635dbebedf60ee7d37c83460295ca9fb64fedc0c2e2ccdddf  guix-build-d7d3bb1bbcc0/output/arm64-apple-darwin/bitcoin-d7d3bb1bbcc0-arm64-apple-darwin.tar.gz
500751e3817551bbdb1377b7947bb666a34d018a018f01b5e4b035e221950f7c  guix-build-d7d3bb1bbcc0/output/dist-archive/bitcoin-d7d3bb1bbcc0.tar.gz
38d0a65d05879713888d405e056ab46e00a1eeebccda8698a2f1bc990d769b2d  guix-build-d7d3bb1bbcc0/output/powerpc64-linux-gnu/SHA256SUMS.part
8d6efeb86bad556ee510ddf3c0fac0c15cb257e884a057dd5f51c92dd4689653  guix-build-d7d3bb1bbcc0/output/powerpc64-linux-gnu/bitcoin-d7d3bb1bbcc0-powerpc64-linux-gnu-debug.tar.gz
3a47e308df8f90253999d58eb98e11e0754c252b736171c0c8f36bbf986c0ace  guix-build-d7d3bb1bbcc0/output/powerpc64-linux-gnu/bitcoin-d7d3bb1bbcc0-powerpc64-linux-gnu.tar.gz
ef09a59e0f71d807aa4c68cc4803ef798852b7710ac46e27b09b58dbec8fd841  guix-build-d7d3bb1bbcc0/output/riscv64-linux-gnu/SHA256SUMS.part
0de69c2c507243f2e0da67092e192edf6bb793aa51cbe9567e2e8271d6b8c249  guix-build-d7d3bb1bbcc0/output/riscv64-linux-gnu/bitcoin-d7d3bb1bbcc0-riscv64-linux-gnu-debug.tar.gz
411ffdb06fbe57543ba2fbf112a43eebb0c33fc55d7be1228c6bc5d94ab6377d  guix-build-d7d3bb1bbcc0/output/riscv64-linux-gnu/bitcoin-d7d3bb1bbcc0-riscv64-linux-gnu.tar.gz
4cc835e7bceff7490010838e454aba643814bc1a35536c1cf2200a8c5a52b9b6  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/SHA256SUMS.part
9d64db19ba634888c773467c249fd85f7cb1b32621715e80a9481b8c2b29121e  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/bitcoin-d7d3bb1bbcc0-x86_64-apple-darwin-unsigned.tar.gz
d0d045dad42c87ca6c39aabb7760e6a5c86ffdef112c7bde62ba25d2b5d28302  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/bitcoin-d7d3bb1bbcc0-x86_64-apple-darwin-unsigned.zip
9642fab24e8b57e47d015ec2faa95482551f1741f190b798a4b89b29a30f2b6e  guix-build-d7d3bb1bbcc0/output/x86_64-apple-darwin/bitcoin-d7d3bb1bbcc0-x86_64-apple-darwin.tar.gz
8737d91c062d41f2b14bc39a60b354d18966f5e5b98f08ff85ce3e0948e3c59a  guix-build-d7d3bb1bbcc0/output/x86_64-linux-gnu/SHA256SUMS.part
0e90253dbda704d0877515cfd95b33a0bf289fe6465002bf975b7f7c34a6779a  guix-build-d7d3bb1bbcc0/output/x86_64-linux-gnu/bitcoin-d7d3bb1bbcc0-x86_64-linux-gnu-debug.tar.gz
26421408b850899fc11d08c1d363160edaf215cbddcce53a292686d54c191657  guix-build-d7d3bb1bbcc0/output/x86_64-linux-gnu/bitcoin-d7d3bb1bbcc0-x86_64-linux-gnu.tar.gz
971cdac198218b377178730b088a4ee35f03df5f053bc426216a8e7e5d3f935f  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/SHA256SUMS.part
057d47f78dbdf1efe0809e1172a3cc86797d925bcd2637e264b6d0b4d3777592  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64-debug.zip
cc558956b3cfb170f6e5db4499a5ce6d41a87c236187b273ba8d9cecfd17def3  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64-setup-unsigned.exe
b01709927b7e68588f05de7ce06a44ab604de6dd208acddc11033141719ee5d2  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64-unsigned.tar.gz
a5f2e295890ea1d62833037128e2cfaa7625838b86af6e3689a943cb69298dd9  guix-build-d7d3bb1bbcc0/output/x86_64-w64-mingw32/bitcoin-d7d3bb1bbcc0-win64.zip
```

hebasto committed:
depends: Override default build type for `libevent`

The `libevent` package defaults to the "Release" build type, which
overrides our per-build-type optimization flags with `-O3`.

To prevent this behavior, set `CMAKE_BUILD_TYPE` to "None", consistent
with how other packages are handled.

hebasto commented:
> This change is not deterministic.

Thanks! Should be fixed now.

hebasto commented:
My Guix build:
```
aarch64
22e26035bc1c8484369d855033af86a1209103e9f5c013db27c9a5e88f798de4  guix-build-d44626a9c2eb/output/aarch64-linux-gnu/SHA256SUMS.part
161a7f07ec0bbd8471453249319f179b6478f18b506b837404d6fd661c3790cf  guix-build-d44626a9c2eb/output/aarch64-linux-gnu/bitcoin-d44626a9c2eb-aarch64-linux-gnu-debug.tar.gz
6288a7dfb39d4816b8e43666b7690db2a7fbbcc9077a4b1ec72ecac9804dd8a6  guix-build-d44626a9c2eb/output/aarch64-linux-gnu/bitcoin-d44626a9c2eb-aarch64-linux-gnu.tar.gz
428e0bf1e4076e3f45d916347488c6748ef26661ac771b30fee3607b50a65e6e  guix-build-d44626a9c2eb/output/arm-linux-gnueabihf/SHA256SUMS.part
eb4bf22841dcd5acfad0d1775e9954c88e32314b92032184ba23727da5f5dcfd  guix-build-d44626a9c2eb/output/arm-linux-gnueabihf/bitcoin-d44626a9c2eb-arm-linux-gnueabihf-debug.tar.gz
e2210ea60ea8f52350b983cc7b151cebed4d495b68276f28c61d57ffb317b706  guix-build-d44626a9c2eb/output/arm-linux-gnueabihf/bitcoin-d44626a9c2eb-arm-linux-gnueabihf.tar.gz
4d857532149b4a99693d7bc0f82917ec8f1b2f66331e2950d8b64c6f59c3f41b  guix-build-d44626a9c2eb/output/arm64-apple-darwin/SHA256SUMS.part
4a7c0b9e1c846e9896939e0a8314a2ff63993939f4b7050de1f99fa904a5c1bd  guix-build-d44626a9c2eb/output/arm64-apple-darwin/bitcoin-d44626a9c2eb-arm64-apple-darwin-unsigned.tar.gz
0264205932f949516767bdeb8b4cd4330fef75fa3ddfeec6ca86b78ad09e9c50  guix-build-d44626a9c2eb/output/arm64-apple-darwin/bitcoin-d44626a9c2eb-arm64-apple-darwin-unsigned.zip
30117d702f11e9334b3d34f6d0779ac77133355f60a9d399669cbd6a6de40565  guix-build-d44626a9c2eb/output/arm64-apple-darwin/bitcoin-d44626a9c2eb-arm64-apple-darwin.tar.gz
c86d2b4a2736bc7d8dc79cbf66757e6073634621a6d7c1fdb19add029273885c  guix-build-d44626a9c2eb/output/dist-archive/bitcoin-d44626a9c2eb.tar.gz
fc4d14b10e50853c86c50b5c14fd46d7cc5fb0e9e083f10d60a5f80135073ae0  guix-build-d44626a9c2eb/output/powerpc64-linux-gnu/SHA256SUMS.part
37a9aae92d653ce41fd9be0dd87f53d532a340bf248faf09aad94b7d07009804  guix-build-d44626a9c2eb/output/powerpc64-linux-gnu/bitcoin-d44626a9c2eb-powerpc64-linux-gnu-debug.tar.gz
8086249bcfbc76ed100271dc6705a3a731f2b2f2990593746002f25b50452f46  guix-build-d44626a9c2eb/output/powerpc64-linux-gnu/bitcoin-d44626a9c2eb-powerpc64-linux-gnu.tar.gz
89c7469244dd861141203970fde4a785917248b6df27184e15be8068df63d782  guix-build-d44626a9c2eb/output/riscv64-linux-gnu/SHA256SUMS.part
e7feaaa3bd62657d101f878563bd38acc00b80941553a04ba25db6b6c28e7696  guix-build-d44626a9c2eb/output/riscv64-linux-gnu/bitcoin-d44626a9c2eb-riscv64-linux-gnu-debug.tar.gz
6417982eec8705dde8dc2e9325e0e92e6648b4d98b8e1f2dcbe6bc183100d37f  guix-build-d44626a9c2eb/output/riscv64-linux-gnu/bitcoin-d44626a9c2eb-riscv64-linux-gnu.tar.gz
64ae0768133cb0a2cca69b0575f28f494eb632ce1fc9c3afe00072e050f617f0  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/SHA256SUMS.part
6d347967fdc8e19bd671879ac38dcae27b85f781e7a789810de039c8e605d4ac  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/bitcoin-d44626a9c2eb-x86_64-apple-darwin-unsigned.tar.gz
20e76be3dfe370e86c3822e2d578e4ab652cff32bc1c16dbbcad0fa93b8d65a6  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/bitcoin-d44626a9c2eb-x86_64-apple-darwin-unsigned.zip
c56a94c7daa806cd9f6f41dbcb39e4d32ad87875de7b44697d134c602253f9c3  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/bitcoin-d44626a9c2eb-x86_64-apple-darwin.tar.gz
7a5c6fd775818905565430d466e9c424804ba6dfc8926a7316b0632b60a71a76  guix-build-d44626a9c2eb/output/x86_64-linux-gnu/SHA256SUMS.part
1c7d7f1b1cb8e46946c0ddb0e95a53dcb1eb3736944fe0136fede484618f6c33  guix-build-d44626a9c2eb/output/x86_64-linux-gnu/bitcoin-d44626a9c2eb-x86_64-linux-gnu-debug.tar.gz
09644322b5eba0bf16f952a8bfa3ca7d1f8f2e5883995bef1f8b9d033eafe61c  guix-build-d44626a9c2eb/output/x86_64-linux-gnu/bitcoin-d44626a9c2eb-x86_64-linux-gnu.tar.gz
ba4725e511dc436b7836a95b7d116b55bc5e275dcea444bf8b85ca060e1c8a3a  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/SHA256SUMS.part
3062b7e99af02a76e349757845eebcd226019eab4edee8e0ca960517166fe9fd  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64-debug.zip
36b20f64806150b4e6f5052341e427edeeb1348ee5c6a5442de2dae632737779  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64-setup-unsigned.exe
30c3bebbc1326f0a2dfc619d9278559ba55bbb8e459cb62ca03780ac31e76694  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64-unsigned.tar.gz
12c334eccc4601322006b779dc4a1f9edd34c787815342da49c9971ea62f4d78  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64.zip
```

fanquake commented:
Guix Build (x86_64):
```bash
22e26035bc1c8484369d855033af86a1209103e9f5c013db27c9a5e88f798de4  guix-build-d44626a9c2eb/output/aarch64-linux-gnu/SHA256SUMS.part
161a7f07ec0bbd8471453249319f179b6478f18b506b837404d6fd661c3790cf  guix-build-d44626a9c2eb/output/aarch64-linux-gnu/bitcoin-d44626a9c2eb-aarch64-linux-gnu-debug.tar.gz
6288a7dfb39d4816b8e43666b7690db2a7fbbcc9077a4b1ec72ecac9804dd8a6  guix-build-d44626a9c2eb/output/aarch64-linux-gnu/bitcoin-d44626a9c2eb-aarch64-linux-gnu.tar.gz
428e0bf1e4076e3f45d916347488c6748ef26661ac771b30fee3607b50a65e6e  guix-build-d44626a9c2eb/output/arm-linux-gnueabihf/SHA256SUMS.part
eb4bf22841dcd5acfad0d1775e9954c88e32314b92032184ba23727da5f5dcfd  guix-build-d44626a9c2eb/output/arm-linux-gnueabihf/bitcoin-d44626a9c2eb-arm-linux-gnueabihf-debug.tar.gz
e2210ea60ea8f52350b983cc7b151cebed4d495b68276f28c61d57ffb317b706  guix-build-d44626a9c2eb/output/arm-linux-gnueabihf/bitcoin-d44626a9c2eb-arm-linux-gnueabihf.tar.gz
4d857532149b4a99693d7bc0f82917ec8f1b2f66331e2950d8b64c6f59c3f41b  guix-build-d44626a9c2eb/output/arm64-apple-darwin/SHA256SUMS.part
4a7c0b9e1c846e9896939e0a8314a2ff63993939f4b7050de1f99fa904a5c1bd  guix-build-d44626a9c2eb/output/arm64-apple-darwin/bitcoin-d44626a9c2eb-arm64-apple-darwin-unsigned.tar.gz
0264205932f949516767bdeb8b4cd4330fef75fa3ddfeec6ca86b78ad09e9c50  guix-build-d44626a9c2eb/output/arm64-apple-darwin/bitcoin-d44626a9c2eb-arm64-apple-darwin-unsigned.zip
30117d702f11e9334b3d34f6d0779ac77133355f60a9d399669cbd6a6de40565  guix-build-d44626a9c2eb/output/arm64-apple-darwin/bitcoin-d44626a9c2eb-arm64-apple-darwin.tar.gz
c86d2b4a2736bc7d8dc79cbf66757e6073634621a6d7c1fdb19add029273885c  guix-build-d44626a9c2eb/output/dist-archive/bitcoin-d44626a9c2eb.tar.gz
fc4d14b10e50853c86c50b5c14fd46d7cc5fb0e9e083f10d60a5f80135073ae0  guix-build-d44626a9c2eb/output/powerpc64-linux-gnu/SHA256SUMS.part
37a9aae92d653ce41fd9be0dd87f53d532a340bf248faf09aad94b7d07009804  guix-build-d44626a9c2eb/output/powerpc64-linux-gnu/bitcoin-d44626a9c2eb-powerpc64-linux-gnu-debug.tar.gz
8086249bcfbc76ed100271dc6705a3a731f2b2f2990593746002f25b50452f46  guix-build-d44626a9c2eb/output/powerpc64-linux-gnu/bitcoin-d44626a9c2eb-powerpc64-linux-gnu.tar.gz
89c7469244dd861141203970fde4a785917248b6df27184e15be8068df63d782  guix-build-d44626a9c2eb/output/riscv64-linux-gnu/SHA256SUMS.part
e7feaaa3bd62657d101f878563bd38acc00b80941553a04ba25db6b6c28e7696  guix-build-d44626a9c2eb/output/riscv64-linux-gnu/bitcoin-d44626a9c2eb-riscv64-linux-gnu-debug.tar.gz
6417982eec8705dde8dc2e9325e0e92e6648b4d98b8e1f2dcbe6bc183100d37f  guix-build-d44626a9c2eb/output/riscv64-linux-gnu/bitcoin-d44626a9c2eb-riscv64-linux-gnu.tar.gz
64ae0768133cb0a2cca69b0575f28f494eb632ce1fc9c3afe00072e050f617f0  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/SHA256SUMS.part
6d347967fdc8e19bd671879ac38dcae27b85f781e7a789810de039c8e605d4ac  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/bitcoin-d44626a9c2eb-x86_64-apple-darwin-unsigned.tar.gz
20e76be3dfe370e86c3822e2d578e4ab652cff32bc1c16dbbcad0fa93b8d65a6  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/bitcoin-d44626a9c2eb-x86_64-apple-darwin-unsigned.zip
c56a94c7daa806cd9f6f41dbcb39e4d32ad87875de7b44697d134c602253f9c3  guix-build-d44626a9c2eb/output/x86_64-apple-darwin/bitcoin-d44626a9c2eb-x86_64-apple-darwin.tar.gz
7a5c6fd775818905565430d466e9c424804ba6dfc8926a7316b0632b60a71a76  guix-build-d44626a9c2eb/output/x86_64-linux-gnu/SHA256SUMS.part
1c7d7f1b1cb8e46946c0ddb0e95a53dcb1eb3736944fe0136fede484618f6c33  guix-build-d44626a9c2eb/output/x86_64-linux-gnu/bitcoin-d44626a9c2eb-x86_64-linux-gnu-debug.tar.gz
09644322b5eba0bf16f952a8bfa3ca7d1f8f2e5883995bef1f8b9d033eafe61c  guix-build-d44626a9c2eb/output/x86_64-linux-gnu/bitcoin-d44626a9c2eb-x86_64-linux-gnu.tar.gz
ba4725e511dc436b7836a95b7d116b55bc5e275dcea444bf8b85ca060e1c8a3a  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/SHA256SUMS.part
3062b7e99af02a76e349757845eebcd226019eab4edee8e0ca960517166fe9fd  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64-debug.zip
36b20f64806150b4e6f5052341e427edeeb1348ee5c6a5442de2dae632737779  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64-setup-unsigned.exe
30c3bebbc1326f0a2dfc619d9278559ba55bbb8e459cb62ca03780ac31e76694  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64-unsigned.tar.gz
12c334eccc4601322006b779dc4a1f9edd34c787815342da49c9971ea62f4d78  guix-build-d44626a9c2eb/output/x86_64-w64-mingw32/bitcoin-d44626a9c2eb-win64.zip
```

fanquake commented on depends/packages/libevent.mk:
```diff
@@ -10,9 +10,10 @@ $(package)_build_subdir=build
+  $(package)_config_opts=-DCMAKE_BUILD_TYPE=None -DEVENT__DISABLE_BENCHMARK=ON -DEVENT__DISABLE_OPENSSL=ON
   $(package)_config_opts+=-DEVENT__DISABLE_SAMPLES=ON -DEVENT__DISABLE_REGRESS=ON
   $(package)_config_opts+=-DEVENT__DISABLE_TESTS=ON -DEVENT__LIBRARY_TYPE=STATIC
+  $(package)_cflags += -ffile-prefix-map=$($(package)_extract_dir)=/usr
```
Something that should probably be followed up with, given that most coverage builds are using depends, is the fact that `-ffile-prefix-map` keeps being hardcoded into depends, while CMakelists.txt says:
```bash
# Avoiding the `-ffile-prefix-map` compiler option because it implies
# `-fcoverage-prefix-map` on Clang or `-fprofile-prefix-map` on GCC,
# which can cause issues with coverage builds,
```
Might be less of an issue given these are deps but it's at least inconsistent.

fanquake replied:
This did break oss-fuzz. See #31770.

hebasto replied:
> Something that should probably be followed up with, given that most coverage builds are using depends, is the fact that `-ffile-prefix-map` keeps being hardcoded into depends, while CMakelists.txt says:
> 
> ```shell
> # Avoiding the `-ffile-prefix-map` compiler option because it implies
> # `-fcoverage-prefix-map` on Clang or `-fprofile-prefix-map` on GCC,
> # which can cause issues with coverage builds,
> ```
> 
> Might be less of an issue given these are deps but it's at least inconsistent.

You're right.

Amended in https://github.com/bitcoin/bitcoin/pull/31800.

fanquake reviewed (approved):
ACK d44626a9c2eb95b434edc7a22a70f2cce63c9d84
